src/main/java/org/jabref/logic/cleanup/MoveFilesCleanup.java

```diff
@@ -24,9 +27,12 @@ public class MoveFilesCleanup implements CleanupJob {
     private final BibDatabaseContext databaseContext;
     private final FilePreferences filePreferences;
 
-    public MoveFilesCleanup(BibDatabaseContext databaseContext, FilePreferences filePreferences) {
+    private final DialogService dialogService;
```

Comment:
Oh, no `gui` in logic.

Please add a List<String> of errors. Add a getter for it. after the call, from the gui, call the getter in the gui and display (sing the dialogservice)

Reply:
@koppor, so should I create an instance of MoveFilesCleanup in gui now?

Reply:
You already have. See src/main/java/org/jabref/gui/externalfiles/ExternalFilesEntryLinker.java.

You need to work on passing back the results.

The cleanup should not worry of how to format the results; that should be done by the caller.